### PR TITLE
Grpc hosting

### PR DESCRIPTION
**Pull request by einari**
This is the fundamental building block for us to use when exposing gRPC services. 
For instance, our runtime will have Application (connected to by SDKs), Interaction (exposing our pipelines), Management (management services).

The reason for pulling this down to fundamentals, is that we need this in more places than just the runtime.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1137957067184597)

### CODE REVIEW

File: Source/Grpc/HostConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Dolittle. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+namespace Dolittle.Grpc
+{
+    /// <summary>
+    /// Represents the configuration for a
```
Comment (woksin) [inline Source/Grpc/HostConfiguration.cs]: Missing parts of the documentation

File: Specifications/Grpc/for_Hosts/given/ICanBindMyHostType.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Dolittle. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+namespace Dolittle.Grpc.given
+{
+    public interface ICanBindMyHostType : ICanBindServices
```
Comment (woksin) [inline Specifications/Grpc/for_Hosts/given/ICanBindMyHostType.cs]: Missing docs

Comment (einari) [reply Specifications/Grpc/for_Hosts/given/ICanBindMyHostType.cs]: We don't put xml docs on classes used in specs - they aren't published anywhere

File: Source/Grpc/HostConfiguration.cs (hunk 1 above)
Comment (einari) [reply Source/Grpc/HostConfiguration.cs]: Hehe.. Will fix

**Pull request merged**